### DISCUSSION TITLE
`of` emits a value after `take(n)` should've already unsubscribed

### DISCUSSION THREAD

**Discussion by OmerGronich** [ReactiveX/rxjs#7366] [Q&A] [answered]
Hello,

I ran into the following behavior which seems like a bug.  

In this stream, I have a take(3) so I expected to see 3 logs from each concatMap.
Why am I seeing a fourth value emission from the first concatMap? 
Important thing to note, if you try to return a synchronous value from the concatMap, like of([v, v]) instead of a promise the fourth value emission disappears which makes me think it has something to do with promises inside observables and schedulers. 

![image](https://github.com/ReactiveX/rxjs/assets/59396491/b3a59926-17ec-4052-a8a4-8015383aba1d)

stackblitz link: 
https://stackblitz.com/edit/rxjs-9cqjn6?file=index.ts

**Comment by voliva** [comment, marked as answer]
I think this is expected:

- `of` emit all of the values synchronously.
- The first `concatMap` keeps all of those emissions on a buffer, as it needs to run the map function for each one of them.
- For each one, it waits for a microtask to complete before moving to the next one.
- The second `concatMap` does the same: It will wait for a microtask to complete before moving to the next one.
- Then `take(3)` will unsubscribe once the second `concatMap` has emitted for the 3rd time.
- But before that happens, the first `concatMap` had values on their buffer, so they will start mapping it before the second `concatMap` emits the 3rd value.

I hope this explains it 🙏
